src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -292,5 +309,127 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            if (!TryReadFileAsSpan(statusFilePath, reusableReader, out ReadOnlySpan<char> statusFileContents))
+            {
+                // Between the time that we get an ID and the time that we try to read the associated stat
+                // file(s), the process could be gone.
+                result = default(ParsedStatus);
+                return false;
+            }
+
+            var results = default(ParsedStatus);
+            while (!statusFileContents.IsEmpty)
+            {
+                int startIndex = statusFileContents.IndexOf(':');
+                int endIndex = statusFileContents.IndexOf('\n');
+
+                // Reached end of file
+                if (endIndex == -1)
```

Comment:
There's always a \n at the end of the file?

Reply:
No. `IndexOf` returns `-1` if  `\n` is not found.

Reply:
Right.  What I was asking was, what if there isn't a \n at the end of the file?  You'll break out here, rather than processing the last segment that was terminated by EOF rather than a \n.